stl/inc/functional

```diff
@@ -29,6 +29,33 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
+#if _HAS_CXX23
+// FUNCTION TEMPLATE invoke_r
+// clang-format off
+template <class _Result_type, class _Callable>
+requires is_invocable_r_v<_Result_type, _Callable>
+_NODISCARD constexpr _Result_type invoke_r(_Callable&& _Obj) noexcept(
+    is_nothrow_invocable_r_v<_Result_type, _Callable>) {
+    if constexpr (is_void_v<_Result_type>) {
+        (void) _STD invoke(_STD forward<_Callable>(_Obj));
+    } else {
+        return _STD invoke(_STD forward<_Callable>(_Obj));
+    }
+}
+
+template <class _Result_type, class _Callable, class _Ty1, class... _Types2>
+requires is_invocable_r_v<_Result_type, _Callable, _Ty1, _Types2...>
+_NODISCARD constexpr _Result_type invoke_r(_Callable&& _Obj, _Ty1&& _Arg1, _Types2&&... _Args2) noexcept(
+    is_nothrow_invocable_r_v<_Result_type, _Callable, _Ty1, _Types2...>) {
+    if constexpr (is_void_v<_Result_type>) {
+        (void) _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Ty1>(_Arg1), _STD forward<_Types2>(_Args2)...);
+    } else {
+        return _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Ty1>(_Arg1), _STD forward<_Types2>(_Args2)...);
+    }
+}

```

Comment:
```suggestion
template <class _Result_type, class _Callable, class... _Types>
requires is_invocable_r_v<_Result_type, _Callable, _Types...>
_NODISCARD constexpr _Result_type invoke_r(_Callable&& _Obj, _Types&&... _Args) noexcept(
    is_nothrow_invocable_r_v<_Result_type, _Callable, _Types...>) {
    if constexpr (is_void_v<_Result_type>) {
        (void) _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Types>(_Args)...);
    } else {
        return _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Types>(_Args)...);
    }
}
```

Reply:
Huh, that was what I first did (though missing void case) and it didn't work so I just copied the `invoke` implementation (failing to copy the single arg case). I want to go back in time and figure out how I messed that up :/